roboteam_ai/src/skills/Keeper.cpp

```diff
@@ -29,33 +31,33 @@ Keeper::Status Keeper::onUpdate() {
 
     if (ball->pos.x < 0) {
         auto attacker = world::world->getRobotClosestToPoint(ball->pos, world::THEIR_ROBOTS);
-        if ((ball->pos - attacker->pos).length() < 0.3) {
+        if (attacker&&(ball->pos - attacker->pos).length() < 0.3) {
             setGoalPosWithAttacker(attacker);
         }
     }
 
     blockPoint = computeBlockPoint(ballPos);
 
-    if (!world::field->pointIsInField(blockPoint, static_cast<float>(Constants::OUT_OF_FIELD_MARGIN()))) {
+    if (!world::field->pointIsInField(blockPoint, -static_cast<float>(Constants::OUT_OF_FIELD_MARGIN()))) {
         blockPoint=goalPos;
         blockPoint.x+=Constants::KEEPER_CENTREGOAL_MARGIN();
         command.w=0;
     }
     else{
-        command.w=(ballPos-blockPoint).angle();
+        command.w=Angle((ballPos-blockPoint).angle()+M_PI_2).getAngle();
```

Comment:
command.w = (ballPos-blockPoint).toAngle() + M_PI_2 does the same thing

Reply:
That is not explicitly limited; this was what I tried first but gives me errors if the first Angle is larger than M_PI_2